.github/cd.yml

```diff
@@ -0,0 +1,119 @@
+name: Push to Docker Hub
+
+on:
+  release:
+    types:
+      [published]
+      # trigger only on new release
+
+jobs:
+  verify_version:
+    name: Verify version
+    runs-on: ubuntu-latest
+    outputs:
+      # export to be used in other jobs
+      version: ${{ steps.get_version_tag.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+
+      - id: get_version_tag
+        name: Get version tag
+        run: |
+          TAG_VER="${GITHUB_REF##*/}"
+          # set as output:
+          echo "version: ${TAG_VER}"
+          echo "version=${TAG_VER}" >> $GITHUB_OUTPUT
+
+      - id: verify_semantic_tag_format
+        name: Verify tag format
+        # format must be compatible with semantic versioning
+        run: |
+          SEMVER_REGEX="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-((?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
```

Comment:
🤯

Reply:
I think we refactored that at some point here https://github.com/ghga-de/gh-action-ci/blob/main/action.yml. I should work like this. I'm quickly checking if our action is compatible with this repository.

Reply:
Did that now.